### PR TITLE
Make pod utils set working directory for periodic jobs when possible.

### PR DESCRIPTION
**Pull request by cjwagner**
If a periodic job specifies git refs to check out via `extra_refs` we should set the working directory of the test container to the root of the repo checkout for the first ref.

Also moves podspec decoration into its own function.
Review this PR by commit. Moving the decoration into a separate function produced a gross diff.

/area prow
/kind bug
/cc @BenTheElder @stevekuznetsov @kargakis

### CODE REVIEW

File: prow/pod-utils/decorate/podspec.go [hunk 1]
```diff
@@ -231,8 +231,16 @@ func ProwJobToPod(pj kube.ProwJob, buildID string) (*v1.Pod, error) {
 		allEnv[entrypoint.JSONConfigEnvVar] = entrypointConfigEnv
 
 		spec.Containers[0].Command = []string{entrypointLocation}
-		if pj.Spec.Type != kube.PeriodicJob {
-			spec.Containers[0].WorkingDir = clone.PathForRefs(CodeMountPath, pj.Spec.Refs)
+		var checkoutRefs *kube.Refs
```
Comment (fejta) [inline prow/pod-utils/decorate/podspec.go]: Wow you did make this a separate commit, strong work!

**Comment by kargakis** [comment]
We should document this change somewhere. @stevekuznetsov has also brought up making periodics work with git refs.

**Comment by cjwagner** [comment]
PTAL

**Comment by cjwagner** [comment]
Thanks for the review Erick. I'll address the doc fixes in a follow up, these changes are needed urgently.
/hold cancel
